### ISSUE TITLE
Click 'Enter the globally unique name for the new Function App' result in undefined function id

### ISSUE THREAD

**Issue by lirenhe** [Microsoft/vscode-azurefunctions#260]
We are using commands to invoke 'createFunctionApp'.
The code we use:
      const functionAppId: string|undefined =
          await vscode.commands.executeCommand<string>(
              'azureFunctions.createFunctionApp', subscriptionId);

The following item would show  
![capture](https://user-images.githubusercontent.com/9100546/37131759-e33cab10-2281-11e8-836b-13998c1c3b05.PNG)

When clicking 'Enter the globally unique name for the new Function App' , the above function would return functionId = undefined

**Comment by EricJizbaMSFT** [comment]
@lirenhe just curious what inspired you to click that? I agree its weird that it closes the inputbox, but I never would've thought to click it in the first place.

Secondly, you should definitely handle the case where we return undefined for the functionAppId (if you're not already). A return value of undefined signifies that the user canceled the operation (usually by hitting the 'escape' key while in the inputbox - not like how you described above).

**Comment by lirenhe** [comment]
@EricJizbaMSFT , this issue comes from our bug bash. I agree that people should not click that but we could not avoid it.
From user experience prospective, the expected behavior should be which it is clicked, that click event should be ignored instead of return 'undefined' function id so that user could still type in the box for the function app name.

**Comment by EricJizbaMSFT** [comment]
Yes I completely agree with you. My question was more related to the priority of the bug and I think you've answered that. Thanks
